src/packages/server/interfaces.js

```diff
@@ -2,9 +2,19 @@
 import type Logger from '../logger';
 import type Router from '../router';
 
+export type Server$cors = {
```

Comment:
This is a good pattern you've got going on here. I think we should keep the actual `Server$config` as a separate type declaration so we can properly isolate what is user-provided and what is required to internally get passed via the framework.

```javascript
export type Server$config = {
  cors: {
    origin?: string;
    enabled: boolean;
    headers?: Array<string>;
    methods?: Array<string>;
  }
};

export type Server$opts = Server$config & {
  logger: Logger;
  router: Router;
};
```

P.S we should always put semicolons after a `type =` declaration. The only instance where this is not necessary would be an `interface` declaration as it is a `statement` rather than an `expression`.